core-bundle/src/Migration/Version500/AbstractGuestsMigration.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Migration\Version500;
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Contao\StringUtil;
+use Doctrine\DBAL\Connection;
+
+/**
+ * This class can be used as a base class for migrations in extensions for
+ * migrating the "show to guests only" setting to the user group.
+ *
+ * @see GuestsMigration for an example how to use this base class.
+ */
+abstract class AbstractGuestsMigration extends AbstractMigration
+{
+    public function __construct(private Connection $connection)
+    {
+    }
+
+    public function shouldRun(): bool
+    {
+        $schemaManager = $this->connection->createSchemaManager();
+
+        foreach ($this->getTables() as $table) {
+            if (!$schemaManager->tablesExist([$table]) || !isset($schemaManager->listTableColumns($table)['guests'])) {
```

Comment:
```suggestion
            if (!$schemaManager->tablesExist([$table]) || !isset($schemaManager->listTableColumns($table)['guests'] || !isset($schemaManager->listTableColumns($table)['groups'])) {
```

Reply:
Changed in 1adcfd7706a4e03e32e645548a0286ef675ce075.